src/app/shared/navigation-focus/navigation-focus.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+import {Event, NavigationEnd, Router} from '@angular/router';
+import {filter} from 'rxjs/operators';
+
+@Injectable()
```

Comment:
I would probably make this a singleton service: https://angular.io/guide/singleton-services#providing-a-singleton-service